Newsletter2Go/Export/Model/Observer/RegisterIntegration.php

```diff
@@ -79,41 +84,23 @@ protected function getToken($currentToken)
      *
      * @throws \Exception
      */
-    protected function createNewToken($token)
+    protected function recreateToken($token)
     {
-        /** @var Session $adminSession */
-        $adminSession = $this->om->get(Session::class);
-        $adminId = $adminSession->getUser()->getData('user_id');
-
-        /** @var IntegrationModel\Oauth\Token $tokenModel */
-        $tokenModel = $this->om->create(IntegrationModel\Oauth\Token::class);
-        $tokenModel->createAdminToken($adminId);
-        $tokenModel->setToken($token);
-        $tokenModel->setCallbackUrl(self::NEWSLETTER2GO_URL);
-        $tokenModel->save();
-    }
-
-    /**
-     * Revokes all previous tokens
-     *
-     * @param $activeToken
-     *
-     * @return int
-     *
-     * @throws \Magento\Framework\Exception\LocalizedException
-     */
-    protected function revokePreviousTokens($activeToken)
-    {
-        /** @var IntegrationModel\ResourceModel\Oauth\Token $resource */
-        $resource = $this->om->get(IntegrationModel\ResourceModel\Oauth\Token::class);
-        $connection = $resource->getConnection();
-        if (!$connection) {
-            throw new Exception(new Phrase('Unable to fetch db connection!'));
-        }
+        /** @var IntegrationModel\Integration $integration */
+        $integration = $this->om->get(IntegrationModel\Integration::class);
+        $integration->load(self::NEWSLETTER2GO_INTEGRATION_NAME, IntegrationModel\Integration::NAME);
+        $integration->setStatus(IntegrationModel\Integration::STATUS_ACTIVE);
+        $integration->save();
```

Comment:
what happens in the case that a user completely delete the token instead of just disablling it?

Reply:
I don't think that is possible from the UI.

Reply:
ok 👍